lib/Listener/TaskProcessingListener.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\Listener;
+
+use OCA\Mail\AppInfo\Application;
+use OCA\Mail\Db\MessageMapper;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\TaskProcessing\Events\TaskSuccessfulEvent;
+use OCP\TaskProcessing\TaskTypes\TextToTextSummary;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @template-implements IEventListener<Event>
+ */
+class TaskProcessingListener implements IEventListener {
+
+	public function __construct(
+		private LoggerInterface $logger,
+		private MessageMapper $messageMapper,
+	) {
+	}
+
+	public function handle(Event $event): void {
+
+		if (!($event instanceof TaskSuccessfulEvent)) {
+			return;
+		}
+
+		$task = $event->getTask();
+
+		if ($task->getAppId() !== Application::APP_ID) {
+			return;
+		}
+
+		if ($task->getTaskTypeId() !== TextToTextSummary::ID) {
+			return;
+		}
+
+		if ($task->getCustomId() !== null) {
+			[$type, $id] = explode(':', $task->getCustomId());
+		} else {
+			$this->logger->info('Error handling task processing event custom id missing');
+			return;
+		}
+		if ($type === null || $id === null) {
+			$this->logger->info('Error handling task processing event custom id is invalid');
+			return;
+		}
+		if ($task->getUserId() !== null) {
+			$userId = $task->getUserId();
+		} else {
+			$this->logger->info('Error handling task processing event user id missing');
+			return;
+		}
+		if ($task->getOutput() !== null) {
+			$output = $task->getOutput();
+			if (isset($task->getOutput()['output'])) {
+				/* @psalm-suppress PossiblyInvalidCast */
```

Comment:
Can you elaborate on the suppressed check?

Reply:
This was my last resort to satisfy psalm and it's still not happy.

![image](https://github.com/user-attachments/assets/6de81616-0335-46ee-a390-1a462be6978e)

Reply:
I stand with Psalm. https://github.com/nextcloud/server/blob/db683da757d854ff0e86d78a8b7053e2677731ac/lib/public/TextProcessing/Task.php#L116-L118 the method returns null or a string. You are interpreting the string as an array. How is that possible? The PHP return type hint won't let `getOutput` return anything but null or string. Other values cause a type error.

Reply:
That's the wrong class... This is the correct one... https://github.com/nextcloud/server/blob/db683da757d854ff0e86d78a8b7053e2677731ac/lib/public/TaskProcessing/Task.php#L155

Anyway, its fixed, psalm is happy